File: segundo/1er cuatri/PDOO/ejercicios_clase/clase25nov/src/clase25nov/Jugador.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package clase25nov;

/**
 *
 * @author alvarogz
 */
public class Jugador extends Personaje {

    public Jugador(String nombre, int vida) {
        super(nombre, vida);
        tipo = TIPO_JUGADOR;
    }
    
    @Override
    public void atacar() {
        System.out.println(getNombre() + " esta atacando con la espada");
    }

    @Override
    public void defender() {
        System.out.println(getNombre() + " se esta defendiendo con su escudo"); 
    }
    
}

File: segundo/1er cuatri/PDOO/ejercicios_clase/clase25nov_ruby/jugador.rb
# frozen_string_literal: true
require_relative 'personaje'

class Jugador < Personaje
  def initialize(nombre, vida)
    super(nombre, vida)
    @tipo = @@TIPO_JUGADOR
  end

  def atacar
    puts "#{@nombre} esta atacando con la espada"
  end

  def defender
    puts "#{@nombre} se esta defendiendo con su escudo"
  end
end
